itest/lib/app.js

```diff
@@ -194,35 +191,66 @@ export const getSearchSpeed = (app: Application) =>
 export const getSearchTime = (app: Application) =>
   getSearchStat(app, selectors.search.time)
 
-export const setSpan = (app: Application, span: string) => {
+export const setSpan = async (app: Application, span: string) => {
   const clickSpanButton = () =>
-    appStep("click span selector", () =>
-      app.client
-        .waitForVisible(selectors.span.button)
-        .then(() => app.client.click(selectors.span.button))
-    )
+    appStep("click span selector", () => click(app, selectors.span.button))
 
   const clickSpan = () =>
     appStep(`select span ${span}`, () =>
-      app.client
-        .waitForVisible(selectors.span.menuItem(span))
-        .then(() => app.client.click(selectors.span.menuItem(span)))
+      click(app, selectors.span.menuItem(span))
     )
 
-  return clickSpanButton().then(() => clickSpan())
+  await clickSpanButton()
+  await clickSpan()
```

Comment:
I don't see the reason that these need to be in-line functions. Any reason not to run the bodies of these functions directly?

Reply:
👍